src/components/subscriptions/licenses/LicenseManagementTable/index.jsx

```diff
@@ -0,0 +1,296 @@
+            break;
+          case 'email':
+            setSearchQuery(filter.value);
+            break;
+          default: break;
+        }
+      });
+    }
+  };
+
+  const debouncedUpdateFilters = debounce(
+    updateFilters,
+    DEBOUNCE_TIME_MILLIS,
+  );
+
+  const debouncedSetCurrentPage = debounce(
+    setCurrentPage,
+    DEBOUNCE_TIME_MILLIS,
+  );
+  // Call back function, handles filters and page changes
+  const fetchData = useCallback(
+    (args) => {
+      // pages index from 1 in backend, Datable component index from 0
+      if (args.pageIndex !== currentPage - 1) {
+        debouncedSetCurrentPage(args.pageIndex + 1);
+      }
+      debouncedUpdateFilters(args.filters);
+    },
+    [currentPage],
+  );
+
+  // Maps user to rows
+  const data = useMemo(
+    () => users?.results?.map(user => ({
+      id: user.uuid,
+      email: user.userEmail,
+      emailLabel: <span data-hj-suppress>{user.userEmail}</span>,
+      status: user.status,
+      statusBadge: <LicenseManagementUserBadge userStatus={user.status} />,
+      recent: userRecentAction(user),
+    })),
+    [users],
+  );
+
+  // Row action button functions
+  const rowRemindOnClick = (remindUser) => {
+    setRemindModal({
+      open: true,
+      users: [remindUser],
+    });
+  };
+  const rowRevokeOnClick = (revokeUser) => {
+    setRevokeModal({
+      open: true,
+      users: [revokeUser],
+    });
+  };
+
+  // Successful action modal callback
+  const onRemindSuccess = () => {
+    forceRefreshUsers();
+    setRemindModal(modalZeroState);
+    addToast(`User${revokeModal.users.length > 1 ? 's' : ''} successfully reminded`);
+  };
+  const onRevokeSuccess = () => {
+    // refresh subscription and user data to get updated revoke count and revoked user list
+    forceRefreshSubscription();
+    forceRefreshUsers();
+    forceRefreshOverview();
+    setRevokeModal(modalZeroState);
+    addToast(`License${revokeModal.users.length > 1 ? 's' : ''} successfully revoked`);
+  };
+
+  // Bulk Action buttons
+  const bulkRemindOnClick = (usersToRemind) => {
+    setRemindModal({
+      open: true,
+      users: usersToRemind,
+    });
+  };
+  const bulkRevokeOnClick = (usersToRevoke) => {
+    setRevokeModal({
+      open: true,
+      users: usersToRevoke,
+    });
+  };
+
+  return (
+    <>
+      {(users.count < 1 && userStatusFilter === null) && <SubscriptionZeroStateMessage /> }
+
+      <DataTable
+        showFiltersInSidebar={showFiltersInSidebar}
+        isFilterable
+        manualFilters
+        isSelectable
+        manualSelectColumn={selectColumn}
+        SelectionStatusComponent={DataTable.ControlledSelectionStatus}
+        defaultColumnValues={{ Filter: TextFilter }}
+        isPaginated
+        manualPagination
+        itemCount={users.count}
+        pageCount={users.numPages || 1}
+        // eslint-disable-next-line no-unused-vars
+        tableActions={(i) => <DownloadCsvButton />}
```

Comment:
Do we need to add in the count of rows that the download table action will apply to, as shown in the Figma spec? Also, looks like the button text is changing from "Download CSV" to "Download all" to more closely indicate that clicking it will download the entire table (without any filters applied).

We might need to track the initial number of rows in the table prior to any filtering since the data (table instance) given by the `tableActions` prop would change after applying filters.

Reply:
Going to defer on this. The ability to select individual rows for CSV download requires a backend change, will add a ticket to add this in the furture.

For now, Changed the button text to `Download all` and updated the Iconography to better resemble Figma spec.